src/cursor/cur_hs.c

```diff
@@ -843,9 +859,16 @@ __curhs_update(WT_CURSOR *cursor)
     /* Connect the tombstone to the update. */
     hs_tombstone->next = hs_upd;
 
-    /* Insert the updates and if we fail, search and try again. */
+    /* Make the updates and if we fail, search and try again. */
     while ((ret = __wt_hs_modify(cbt, hs_tombstone)) == WT_RESTART) {
-        WT_WITH_PAGE_INDEX(session, ret = __wt_hs_row_search(cbt, &file_cursor->key, true));
+        WT_WITH_PAGE_INDEX(session, ret = __wt_hs_row_search(cbt, &file_cursor->key, false));
+        WT_ERR(ret);
+        retry = true;
+    }
+
+    /* If we retry, search again to point to the updated value. */
+    if (retry) {
+        WT_WITH_PAGE_INDEX(session, ret = __wt_hs_row_search(cbt, &file_cursor->key, false));
         WT_ERR(ret);
```

Comment:
This will overwrite the error code, I guess we want to return the error from __wt_hs_modify, this error code we need to capture in a temporary error variable.

Reply:
OK